store.go

```diff
@@ -1758,22 +1823,15 @@ func (s *store) ImageSize(id string) (int64, error) {
 }
 
 func (s *store) ContainerSize(id string) (int64, error) {
-	lstore, err := s.LayerStore()
+	lstore, err := s.RWLayerStore()
 	if err != nil {
 		return -1, err
 	}
-	lstores, err := s.ROLayerStores()
-	if err != nil {
-		return -1, err
-	}
-	for _, s := range append([]ROLayerStore{lstore}, lstores...) {
-		store := s
-		store.RLock()
-		defer store.Unlock()
-		if modified, err := store.Modified(); modified || err != nil {
-			if err = store.Load(); err != nil {
-				return -1, err
-			}
+	lstore.RLock()
+	defer lstore.Unlock()
+	if modified, err := lstore.Modified(); modified || err != nil {
```

Comment:
I'd reverse the order of checks here, ie
```suggestion
	if modified, err := lstore.Modified(); err != nil || modified {
```

as you're not supposed to check `modified` in case `err != nil`.

Reply:
Fixed :+1: